pkg/ini/ini.go

```diff
@@ -0,0 +1,203 @@
+package ini
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// Key holds a single INI section and key.
+type Key struct {
+	// Section is an INI section for this key
+	Section string
+	// Name is the key name of an INI setting
+	Name string
+}
+
+// Item holds the value for a Key.
+type Item struct {
+	// Key is the section and key name for this INI setting
+	Key Key
+	// Value is the value for this Key, or empty string
+	Value string
+}
+
+// GetKey returns the value for given INI Key.
+func GetKey(iniFile string, key Key) (Item, error) {
+	fh, err := os.Open(iniFile)
+	if err != nil {
+		return Item{}, fmt.Errorf("failed to open ini file %q: %s", iniFile, err)
+	}
+
+	defer fh.Close()
+
+	scanner := bufio.NewScanner(fh)
+	scanner.Split(bufio.ScanLines)
+
+	var (
+		currentSection string
+		multiline      []string
+	)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if len(multiline) > 0 {
+			if !isMultiline(line) {
+				return Item{
+					Key:   key,
+					Value: strings.Join(multiline, "\n"),
+				}, nil
+			}
+
+			multiline = append(multiline, line)
+
+			continue
+		}
+
+		if isSection(line) {
+			currentSection = getSectionName(line)
+			continue
+		}
+
+		if currentSection != key.Section {
+			continue
+		}
+
+		split := strings.SplitN(line, "=", 2)
+		if len(split) != 2 {
+			continue
+		}
+
+		possibleKey := getPossibleKeyName(split)
+		if possibleKey != key.Name {
+			continue
+		}
+
+		multiline = append(multiline, strings.TrimLeft(strings.TrimLeft(split[1], " "), "\t"))
+	}
+
+	// key not found in INI file, return empty string
+	return Item{
+		Key:   key,
+		Value: "",
+	}, nil
+}
+
+// SetKey saves the value for given INI Key.
+func SetKey(iniFile string, value Item) error {
+	return SetKeys(iniFile, []Item{value})
+}
+
+// SetKeys saves multiple values for given INI Keys.
+func SetKeys(iniFile string, values []Item) error {
+	if len(values) == 0 {
+		return nil
+	}
+
+	fh, err := os.Open(iniFile)
+	if err != nil {
+		return fmt.Errorf("failed to open ini file %q: %s", iniFile, err)
+	}
+
```

Comment:
I can't use `defer fh.Close()` because we manually call `Close()` before re-opening the file to write it's new contents. `defer` isn't necessary here because we only have a single `return` at the end of the function.